### PR TITLE
feature/conversation-screen

### PR DESCRIPTION
**Pull request by imranariffin**
#28 

* Demo phone number: https://receive-smss.com/sms/19709303781/
* Bugs noted:
* * sometimes send message does not work (as seen in demo below)
* * the send button icon glitches upon clicking

Demo (wait for last message "Yay"):
hooligram-global-chat-first-demo

### CODE REVIEW

**Comment by imranariffin** [comment]
@cglotr, this PR is ready for review.

File: app/hooligram-api/hooligramApi.js [hunk 1]
```diff
@@ -29,6 +30,14 @@ const hooligramApi = (config) => (store) => {
     catch (err) {
       action = websocketError(err)
     }
+
+    // hack since backend currently does not provide id for each message
+    // I believe it's the responsibility of backend to generate and provide
+    // the message id, but we can think and discuss about it
```
Comment (cglotr) [inline app/hooligram-api/hooligramApi.js]: Yeah, let's include this to the agenda

File: app/state/actions/authorization.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export const authorizationSignInRequest = (code, country_code, phone_number) =>
+  ({
+    type: API_AUTHORIZATION_SIGN_IN_REQUEST,
+    payload: {
+      code,
+      country_code,
+      phone_number
+    }
+  })
```
Comment (cglotr) [inline app/state/actions/authorization.js]: First time seeing this syntax ahaha

File: app/state/middlewares/navigation/navigation.js [hunk 3]
```diff
@@ -15,17 +15,16 @@ const middleware = navigationActions => store => next => action => {
     return next(action)
   }
 
-  const {
-    routeName: currentRouteName
-  } = navigator.state.nav.routes[navigator.state.nav.index]
+  const route = navigator.state.nav.routes[navigator.state.nav.index]
+  const fullRouteName = `${route.routeName}.${route.routes[route.index].routeName}`
```
Comment (cglotr) [inline app/state/middlewares/navigation/navigation.js]: Is `${route.routeName}.${route.routes[route.index].routeName}` a hack or a standard way of doing it?

File: app/state/reducers/authorization/authorization.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { API_AUTHORIZATION_SIGN_IN_SUCCESS } from '@state/actions/authorization'
+
+export const initialState = {
+  token: {}
```
[the remark this answers is not in the document]
Comment (imranariffin) [reply app/state/reducers/authorization/authorization.js]: Totally agreed on the first point.

On the second ... hmm I did that more for convenience actually. Maybe we should do

```
state = {
  token: '', // == code
  country_code: '',
  phone_number: ''
}
```
instead?

File: app/state/middlewares/navigation/navigation.js (hunk 3 above)
Comment (imranariffin) [reply app/state/middlewares/navigation/navigation.js]: Not a standard. So far it works but probably we can use a better format like `/a/b/c`

Comment (imranariffin) [reply app/state/middlewares/navigation/navigation.js]: Use slash `/` as routeNames delimiter a5971f9

File: app/state/reducers/authorization/authorization.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (imranariffin) [reply app/state/reducers/authorization/authorization.js]: Use above shape for `authorization` state ba90a44

**Pull request merged**